docs/Makefile

```
@@ -1,67 +1,76 @@
 SPHINXOPTS    ?=
 SPHINXBUILD   ?= sphinx-build
-SOURCEDIR     = .
-BUILDDIR      = _build
-VENV          = .sphinx/venv/bin/activate
+SOURCEDIR     ?= .
+BUILDDIR      ?= _build
+VENVDIR       ?= .sphinx/venv
+IN_VENV       ?= . $(VENVDIR)/bin/activate
 
 
 # Put it first so that "make" without argument is like "make help".
 help:
-	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+	@echo "The following make targets are available:"
+	@echo "* install   -- set up a virtual-env to work on documentation"
+	@echo "* run       -- watch, build and serve the documentation"
+	@echo "* html      -- build HTML output under $(BUILDDIR)/html"
+	@echo "* serve     -- serve HTML output from $(BUILDDIR)/html"
+	@echo "* epub      -- build epub output under $(BUILDDIR)/epub"
+	@echo "* pdf       -- build pdf output under $(BUILDDIR)/pdf"
+	@echo "* clean-doc -- clean built doc files"
+	@echo "* clean     -- clean full environment"
+	@echo "* spelling  -- check spelling"
+	@echo "* linkcheck -- check external links"
+	@echo "* woke      -- check inclusive language"
+	@echo "-------------------------------------------"
 
 install:
 	@echo "... setting up virtualenv"
-	python3 -m venv .sphinx/venv
-	. $(VENV); pip install --upgrade -r .sphinx/requirements.txt
-
-	@echo "\n" \
-		"--------------------------------------------------------------- \n" \
-		"* watch, build and serve the documentation: make run \n" \
-                "* only build: make html \n" \
-                "* only epub: make epub \n" \
-                "* only serve: make serve \n" \
-                "* clean built doc files: make clean-doc \n" \
-                "* clean full environment: make clean \n" \
-            		"* check spelling: make spelling \n" \
-                "* check links: make linkcheck \n" \
-                "* check inclusive language: make woke \n" \
-		"--------------------------------------------------------------- \n"
+	python3 -m venv $(VENVDIR)
+	$(IN_VENV); pip install --upgrade -r .sphinx/requirements.txt
+
 run:
-	. $(VENV); sphinx-autobuild -c . -b dirhtml "$(SOURCEDIR)" "$(BUILDDIR)"
+	$(IN_VENV); sphinx-autobuild -b html "$(SOURCEDIR)" "$(BUILDDIR)"/html
 
 html:
-	. $(VENV); $(SPHINXBUILD) -c . -b dirhtml "$(SOURCEDIR)" "$(BUILDDIR)" -w .sphinx/warnings.txt
+	$(IN_VENV); $(SPHINXBUILD) -b html "$(SOURCEDIR)" "$(BUILDDIR)"/html
 
 epub:
-	. $(VENV); $(SPHINXBUILD) -c . -b epub "$(SOURCEDIR)" "$(BUILDDIR)" -w .sphinx/warnings.txt
+	$(IN_VENV); $(SPHINXBUILD) -b epub "$(SOURCEDIR)" "$(BUILDDIR)"/epub
+
+pdf:
+	$(IN_VENV); $(SPHINXBUILD) -b latex "$(SOURCEDIR)" "$(BUILDDIR)"/pdf
+	$(IN_VENV); $(MAKE) -C "$(BUILDDIR)"/pdf all-pdf
 
 serve:
-	cd "$(BUILDDIR)"; python3 -m http.server 8000
+	python3 -m http.server 8000 -d "$(BUILDDIR)"/html
 
 clean: clean-doc
-	rm -rf .sphinx/venv
+	rm -rf "$(VENVDIR)"
 
 clean-doc:
 	git clean -fx "$(BUILDDIR)"
 
+distclean:
+	rm -rf "$(BUILDDIR)"
+	rm -rf "$(VENVDIR)"
+
```

suggestion: reuse `clean-doc` and `clean` target

```suggestion
distclean: clean-doc clean
```

---

Unfortunately this is not really possible; the `clean-doc` target uses `git clean` but when packaging it's not dealing with a git repo, but an unpacked tar-ball so this cannot work.

---

Oh... For some reason I did not register that `clean-doc` has a completely different command. Never mind.